horizomer/utils/tree.py

```diff
@@ -13,6 +13,119 @@
 #
 
 from skbio import TreeNode
+from math import isclose
+from skbio.tree import MissingNodeError
+
+
+def _compare_length(node1, node2):
+    """Private function for compare_branch_lengths. Determines if lengths of the
+    two nodes are same.
+
+    Parameters
+    ----------
+    node1: skbio.TreeNode
+        first node to compare
+    node2: skbio.TreeNode
+        second node to compare
+
+    Returns
+    -------
+    bool
+        `True` if lengths of the two input nodes are None, same, or close
+        `False` otherwise
+
+    See Also
+    --------
+    compare_branch_length
+
+    Examples
+    --------
+    >>> from skbio import TreeNode
+    >>> tree = TreeNode.read(["(a:1,b:1)c;"])
+    >>> print(_compare_length(tree.find('a'), tree.find('b')))
+    True
+    >>> print(_compare_length(tree.find('a'), tree.find('c')))
+    False
+
+    """
+    if node1.length is None and node2.length is None:
```

Comment:
can't you just say `return node1.length == node2.length` ?

Reply:
`isclose()` is used here to handle float.